core/src/core/block.rs

```diff
@@ -397,6 +400,59 @@ impl BlockHeader {
 	}
 }
 
+impl From<UntrustedBlockHeader> for BlockHeader {
+	fn from(header: UntrustedBlockHeader) -> Self {
+		header.0
+	}
+}
+
+/// Block header which does lightweight validation as part of deserialization,
+/// it supposed to be used when we can't trust the channel (eg network)
+pub struct UntrustedBlockHeader(BlockHeader);
+
+/// Deserialization of an untrusted block header
+impl Readable for UntrustedBlockHeader {
+	fn read(reader: &mut dyn Reader) -> Result<UntrustedBlockHeader, ser::Error> {
```

Comment:
Wondering if we never actually pass an `UntrustedBlockHeader` around what this might look like if we modified this signature to actually return the trusted version once we had successfully read it - 

```
impl Readable for UntrustedBlockHeader {
	fn read(reader: &mut dyn Reader) -> Result<BlockHeader, ser::Error> {
        ...
        }
    ...
}
```

I assume we cannot do this easily because of how `Readable` is defined but it would potentially allow us to not need the additonal explicit `.into()` to do conversions later.

Maybe being explicit like this is better though.